core: fix failing Notary test

Sometimes `TestNotary` is failing, and it's probably not due to the insufficient waiting time. This test is complicated, so maybe we need to refactor it.

The problem:
```
2022-04-05T07:56:52.9008397Z === RUN   TestNotary
2022-04-05T07:56:52.9009079Z     logger.go:130: 2022-04-05T07:56:30.546Z	INFO	initial gas supply is not set or wrong, setting default value	{"InitialGASSupply": "52000000"}
2022-04-05T07:56:52.9009879Z     logger.go:130: 2022-04-05T07:56:30.546Z	INFO	mempool size is not set or wrong, setting default value	{"MemPoolSize": 50000}
2022-04-05T07:56:52.9010718Z     logger.go:130: 2022-04-05T07:56:30.546Z	INFO	P2PNotaryRequestPayloadPool size is not set or wrong, setting default value	{"P2PNotaryRequestPayloadPoolSize": 1000}
2022-04-05T07:56:52.9011339Z     logger.go:130: 2022-04-05T07:56:30.546Z	INFO	MaxBlockSize is not set or wrong, setting default value	{"MaxBlockSize": 262144}
2022-04-05T07:56:52.9012009Z     logger.go:130: 2022-04-05T07:56:30.546Z	INFO	MaxBlockSystemFee is not set or wrong, setting default value	{"MaxBlockSystemFee": 900000000000}
2022-04-05T07:56:52.9012703Z     logger.go:130: 2022-04-05T07:56:30.546Z	INFO	MaxTransactionsPerBlock is not set or wrong, using default value	{"MaxTransactionsPerBlock": 512}
2022-04-05T07:56:52.9013443Z     logger.go:130: 2022-04-05T07:56:30.546Z	INFO	MaxValidUntilBlockIncrement is not set or wrong, using default value	{"MaxValidUntilBlockIncrement": 86400}
2022-04-05T07:56:52.9013935Z     logger.go:130: 2022-04-05T07:56:30.546Z	INFO	NativeActivations are not set, using default values
2022-04-05T07:56:52.9014426Z     logger.go:130: 2022-04-05T07:56:30.552Z	INFO	no storage version found! creating genesis block
2022-04-05T07:56:52.9016212Z     logger.go:130: 2022-04-05T07:56:30.554Z	INFO	starting notary service
2022-04-05T07:56:52.9016989Z     logger.go:130: 2022-04-05T07:56:30.555Z	DEBUG	done processing headers	{"headerIndex": 1, "blockHeight": 0, "took": "500.906µs"}
2022-04-05T07:56:52.9017666Z     logger.go:130: 2022-04-05T07:56:31.554Z	INFO	persisted to disk	{"blocks": 1, "keys": 136, "headerHeight": 1, "blockHeight": 1, "took": "112.401µs"}
2022-04-05T07:56:52.9018281Z     logger.go:130: 2022-04-05T07:56:32.781Z	DEBUG	done processing headers	{"headerIndex": 2, "blockHeight": 1, "took": "497.006µs"}
2022-04-05T07:56:52.9018895Z     logger.go:130: 2022-04-05T07:56:32.783Z	DEBUG	done processing headers	{"headerIndex": 3, "blockHeight": 2, "took": "491.205µs"}
2022-04-05T07:56:52.9019554Z     logger.go:130: 2022-04-05T07:56:32.783Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9020210Z     logger.go:130: 2022-04-05T07:56:32.783Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9020834Z     logger.go:130: 2022-04-05T07:56:32.784Z	DEBUG	done processing headers	{"headerIndex": 4, "blockHeight": 3, "took": "555.006µs"}
2022-04-05T07:56:52.9021432Z     logger.go:130: 2022-04-05T07:56:32.838Z	DEBUG	done processing headers	{"headerIndex": 5, "blockHeight": 4, "took": "512.906µs"}
2022-04-05T07:56:52.9022083Z     logger.go:130: 2022-04-05T07:56:32.838Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9022692Z     logger.go:130: 2022-04-05T07:56:32.839Z	DEBUG	done processing headers	{"headerIndex": 6, "blockHeight": 5, "took": "771.108µs"}
2022-04-05T07:56:52.9023298Z     logger.go:130: 2022-04-05T07:56:32.893Z	DEBUG	done processing headers	{"headerIndex": 7, "blockHeight": 6, "took": "499.306µs"}
2022-04-05T07:56:52.9023890Z     logger.go:130: 2022-04-05T07:56:32.894Z	DEBUG	done processing headers	{"headerIndex": 8, "blockHeight": 7, "took": "498.705µs"}
2022-04-05T07:56:52.9024546Z     logger.go:130: 2022-04-05T07:56:32.894Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9025199Z     logger.go:130: 2022-04-05T07:56:32.894Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9025849Z     logger.go:130: 2022-04-05T07:56:32.894Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9026495Z     logger.go:130: 2022-04-05T07:56:32.894Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9027318Z     logger.go:130: 2022-04-05T07:56:32.894Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9027972Z     logger.go:130: 2022-04-05T07:56:32.895Z	DEBUG	done processing headers	{"headerIndex": 9, "blockHeight": 8, "took": "495.505µs"}
2022-04-05T07:56:52.9028590Z     logger.go:130: 2022-04-05T07:56:32.896Z	DEBUG	done processing headers	{"headerIndex": 10, "blockHeight": 9, "took": "540.506µs"}
2022-04-05T07:56:52.9029241Z     logger.go:130: 2022-04-05T07:56:32.896Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9029883Z     logger.go:130: 2022-04-05T07:56:32.896Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9030501Z     logger.go:130: 2022-04-05T07:56:32.897Z	DEBUG	done processing headers	{"headerIndex": 11, "blockHeight": 10, "took": "509.806µs"}
2022-04-05T07:56:52.9031112Z     logger.go:130: 2022-04-05T07:56:32.898Z	DEBUG	done processing headers	{"headerIndex": 12, "blockHeight": 11, "took": "503.705µs"}
2022-04-05T07:56:52.9031710Z     logger.go:130: 2022-04-05T07:56:32.899Z	DEBUG	done processing headers	{"headerIndex": 13, "blockHeight": 12, "took": "630.007µs"}
2022-04-05T07:56:52.9032485Z     logger.go:130: 2022-04-05T07:56:32.897Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9033132Z     logger.go:130: 2022-04-05T07:56:32.900Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9033779Z     logger.go:130: 2022-04-05T07:56:32.900Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9034421Z     logger.go:130: 2022-04-05T07:56:32.900Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9035072Z     logger.go:130: 2022-04-05T07:56:32.900Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9035716Z     logger.go:130: 2022-04-05T07:56:32.900Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9036345Z     logger.go:130: 2022-04-05T07:56:32.900Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9036989Z     logger.go:130: 2022-04-05T07:56:32.900Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9037606Z     logger.go:130: 2022-04-05T07:56:32.901Z	DEBUG	done processing headers	{"headerIndex": 14, "blockHeight": 13, "took": "555.206µs"}
2022-04-05T07:56:52.9038248Z     logger.go:130: 2022-04-05T07:56:32.901Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9038896Z     logger.go:130: 2022-04-05T07:56:32.902Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9039510Z     logger.go:130: 2022-04-05T07:56:32.902Z	DEBUG	done processing headers	{"headerIndex": 15, "blockHeight": 14, "took": "565.907µs"}
2022-04-05T07:56:52.9040158Z     logger.go:130: 2022-04-05T07:56:32.903Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9040801Z     logger.go:130: 2022-04-05T07:56:32.903Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9041412Z     logger.go:130: 2022-04-05T07:56:32.904Z	DEBUG	done processing headers	{"headerIndex": 16, "blockHeight": 15, "took": "567.306µs"}
2022-04-05T07:56:52.9042053Z     logger.go:130: 2022-04-05T07:56:32.904Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9042783Z     logger.go:130: 2022-04-05T07:56:32.904Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9043421Z     logger.go:130: 2022-04-05T07:56:32.905Z	DEBUG	done processing headers	{"headerIndex": 17, "blockHeight": 16, "took": "567.006µs"}
2022-04-05T07:56:52.9044071Z     logger.go:130: 2022-04-05T07:56:32.905Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9044710Z     logger.go:130: 2022-04-05T07:56:32.906Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9045318Z     logger.go:130: 2022-04-05T07:56:32.907Z	DEBUG	done processing headers	{"headerIndex": 18, "blockHeight": 17, "took": "565.806µs"}
2022-04-05T07:56:52.9045960Z     logger.go:130: 2022-04-05T07:56:32.907Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9046608Z     logger.go:130: 2022-04-05T07:56:32.907Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9047219Z     logger.go:130: 2022-04-05T07:56:32.908Z	DEBUG	done processing headers	{"headerIndex": 19, "blockHeight": 18, "took": "578.006µs"}
2022-04-05T07:56:52.9047859Z     logger.go:130: 2022-04-05T07:56:32.908Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9048640Z     logger.go:130: 2022-04-05T07:56:32.909Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9102360Z     logger.go:130: 2022-04-05T07:56:32.909Z	DEBUG	done processing headers	{"headerIndex": 20, "blockHeight": 19, "took": "669.107µs"}
2022-04-05T07:56:52.9103234Z     logger.go:130: 2022-04-05T07:56:32.910Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9103991Z     logger.go:130: 2022-04-05T07:56:32.910Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9104652Z     logger.go:130: 2022-04-05T07:56:32.911Z	DEBUG	done processing headers	{"headerIndex": 21, "blockHeight": 20, "took": "569.806µs"}
2022-04-05T07:56:52.9105336Z     logger.go:130: 2022-04-05T07:56:32.911Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9106023Z     logger.go:130: 2022-04-05T07:56:32.911Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9107444Z     logger.go:130: 2022-04-05T07:56:32.912Z	DEBUG	done processing headers	{"headerIndex": 22, "blockHeight": 21, "took": "560.206µs"}
2022-04-05T07:56:52.9108748Z     logger.go:130: 2022-04-05T07:56:32.913Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9109662Z     logger.go:130: 2022-04-05T07:56:32.913Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9110343Z     logger.go:130: 2022-04-05T07:56:32.913Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9111006Z     logger.go:130: 2022-04-05T07:56:32.913Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9111677Z     logger.go:130: 2022-04-05T07:56:32.913Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9112331Z     logger.go:130: 2022-04-05T07:56:32.913Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9112982Z     logger.go:130: 2022-04-05T07:56:32.913Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9113896Z     logger.go:130: 2022-04-05T07:56:32.913Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9114591Z     logger.go:130: 2022-04-05T07:56:32.914Z	DEBUG	done processing headers	{"headerIndex": 23, "blockHeight": 22, "took": "575.506µs"}
2022-04-05T07:56:52.9115244Z     logger.go:130: 2022-04-05T07:56:32.914Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9115902Z     logger.go:130: 2022-04-05T07:56:32.914Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9116554Z     logger.go:130: 2022-04-05T07:56:32.914Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9117205Z     logger.go:130: 2022-04-05T07:56:32.914Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9117854Z     logger.go:130: 2022-04-05T07:56:32.915Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9118503Z     logger.go:130: 2022-04-05T07:56:32.915Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9119150Z     logger.go:130: 2022-04-05T07:56:32.915Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9119941Z     logger.go:130: 2022-04-05T07:56:32.915Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9120568Z     logger.go:130: 2022-04-05T07:56:32.915Z	DEBUG	done processing headers	{"headerIndex": 24, "blockHeight": 23, "took": "566.506µs"}
2022-04-05T07:56:52.9121216Z     logger.go:130: 2022-04-05T07:56:32.916Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9121870Z     logger.go:130: 2022-04-05T07:56:32.916Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9122519Z     logger.go:130: 2022-04-05T07:56:32.916Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9123147Z     logger.go:130: 2022-04-05T07:56:32.916Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9123800Z     logger.go:130: 2022-04-05T07:56:32.916Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9124447Z     logger.go:130: 2022-04-05T07:56:32.916Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9125094Z     logger.go:130: 2022-04-05T07:56:32.916Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9125747Z     logger.go:130: 2022-04-05T07:56:32.916Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9126391Z     logger.go:130: 2022-04-05T07:56:32.917Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9127038Z     logger.go:130: 2022-04-05T07:56:32.917Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9127683Z     logger.go:130: 2022-04-05T07:56:32.917Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9128325Z     logger.go:130: 2022-04-05T07:56:32.917Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9128969Z     logger.go:130: 2022-04-05T07:56:32.917Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9129704Z     logger.go:130: 2022-04-05T07:56:32.917Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9130376Z     logger.go:130: 2022-04-05T07:56:32.917Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9131025Z     logger.go:130: 2022-04-05T07:56:32.917Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9131647Z     logger.go:130: 2022-04-05T07:56:32.918Z	DEBUG	done processing headers	{"headerIndex": 25, "blockHeight": 24, "took": "575.706µs"}
2022-04-05T07:56:52.9132294Z     logger.go:130: 2022-04-05T07:56:32.918Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9132937Z     logger.go:130: 2022-04-05T07:56:32.918Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9133590Z     logger.go:130: 2022-04-05T07:56:32.918Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9134233Z     logger.go:130: 2022-04-05T07:56:32.918Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9135278Z     logger.go:130: 2022-04-05T07:56:32.918Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9136282Z     logger.go:130: 2022-04-05T07:56:32.918Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9136951Z     logger.go:130: 2022-04-05T07:56:32.918Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9137940Z     logger.go:130: 2022-04-05T07:56:32.918Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9138637Z     logger.go:130: 2022-04-05T07:56:32.920Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9139286Z     logger.go:130: 2022-04-05T07:56:32.920Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9139936Z     logger.go:130: 2022-04-05T07:56:32.920Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9140581Z     logger.go:130: 2022-04-05T07:56:32.920Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9141226Z     logger.go:130: 2022-04-05T07:56:32.920Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9141871Z     logger.go:130: 2022-04-05T07:56:32.920Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9142521Z     logger.go:130: 2022-04-05T07:56:32.920Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9143164Z     logger.go:130: 2022-04-05T07:56:32.920Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9143796Z     logger.go:130: 2022-04-05T07:56:32.920Z	DEBUG	done processing headers	{"headerIndex": 26, "blockHeight": 25, "took": "1.322415ms"}
2022-04-05T07:56:52.9144436Z     logger.go:130: 2022-04-05T07:56:32.922Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9145062Z     logger.go:130: 2022-04-05T07:56:32.922Z	DEBUG	done processing headers	{"headerIndex": 27, "blockHeight": 26, "took": "903.11µs"}
2022-04-05T07:56:52.9145712Z     logger.go:130: 2022-04-05T07:56:32.923Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9150856Z     logger.go:130: 2022-04-05T07:56:32.923Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9151704Z     logger.go:130: 2022-04-05T07:56:32.923Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9152376Z     logger.go:130: 2022-04-05T07:56:32.923Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9153024Z     logger.go:130: 2022-04-05T07:56:32.923Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9153669Z     logger.go:130: 2022-04-05T07:56:32.923Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9154297Z     logger.go:130: 2022-04-05T07:56:32.923Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9154949Z     logger.go:130: 2022-04-05T07:56:32.923Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9155598Z     logger.go:130: 2022-04-05T07:56:32.923Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9156392Z     logger.go:130: 2022-04-05T07:56:32.923Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9157035Z     logger.go:130: 2022-04-05T07:56:32.923Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9157679Z     logger.go:130: 2022-04-05T07:56:32.923Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9158321Z     logger.go:130: 2022-04-05T07:56:32.923Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9158967Z     logger.go:130: 2022-04-05T07:56:32.923Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9159607Z     logger.go:130: 2022-04-05T07:56:32.923Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9160249Z     logger.go:130: 2022-04-05T07:56:32.923Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9160886Z     logger.go:130: 2022-04-05T07:56:32.923Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9161525Z     logger.go:130: 2022-04-05T07:56:32.923Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9162165Z     logger.go:130: 2022-04-05T07:56:32.923Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9162805Z     logger.go:130: 2022-04-05T07:56:32.924Z	DEBUG	done processing headers	{"headerIndex": 28, "blockHeight": 27, "took": "496.305µs"}
2022-04-05T07:56:52.9163448Z     logger.go:130: 2022-04-05T07:56:33.078Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9187535Z     logger.go:130: 2022-04-05T07:56:33.079Z	DEBUG	done processing headers	{"headerIndex": 29, "blockHeight": 28, "took": "586.906µs"}
2022-04-05T07:56:52.9205347Z     logger.go:130: 2022-04-05T07:56:33.080Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9206357Z     logger.go:130: 2022-04-05T07:56:33.080Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9207239Z     logger.go:130: 2022-04-05T07:56:33.080Z	DEBUG	done processing headers	{"headerIndex": 30, "blockHeight": 29, "took": "594.107µs"}
2022-04-05T07:56:52.9208556Z     logger.go:130: 2022-04-05T07:56:33.081Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9209515Z     logger.go:130: 2022-04-05T07:56:33.081Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9210385Z     logger.go:130: 2022-04-05T07:56:33.082Z	DEBUG	done processing headers	{"headerIndex": 31, "blockHeight": 30, "took": "561.306µs"}
2022-04-05T07:56:52.9211255Z     logger.go:130: 2022-04-05T07:56:33.082Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9212113Z     logger.go:130: 2022-04-05T07:56:33.082Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9212942Z     logger.go:130: 2022-04-05T07:56:33.083Z	DEBUG	done processing headers	{"headerIndex": 32, "blockHeight": 31, "took": "559.806µs"}
2022-04-05T07:56:52.9213820Z     logger.go:130: 2022-04-05T07:56:33.083Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9214817Z     logger.go:130: 2022-04-05T07:56:33.083Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9216520Z     logger.go:130: 2022-04-05T07:56:33.084Z	DEBUG	done processing headers	{"headerIndex": 33, "blockHeight": 32, "took": "549.806µs"}
2022-04-05T07:56:52.9217358Z     logger.go:130: 2022-04-05T07:56:33.085Z	DEBUG	done processing headers	{"headerIndex": 34, "blockHeight": 33, "took": "487.506µs"}
2022-04-05T07:56:52.9218252Z     logger.go:130: 2022-04-05T07:56:33.085Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9219111Z     logger.go:130: 2022-04-05T07:56:33.085Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9219945Z     logger.go:130: 2022-04-05T07:56:33.086Z	DEBUG	done processing headers	{"headerIndex": 35, "blockHeight": 34, "took": "502.705µs"}
2022-04-05T07:56:52.9220818Z     logger.go:130: 2022-04-05T07:56:33.086Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9222292Z     logger.go:130: 2022-04-05T07:56:33.086Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9223368Z     logger.go:130: 2022-04-05T07:56:33.086Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9224366Z     logger.go:130: 2022-04-05T07:56:33.086Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9225365Z     logger.go:130: 2022-04-05T07:56:33.087Z	DEBUG	done processing headers	{"headerIndex": 36, "blockHeight": 35, "took": "520.906µs"}
2022-04-05T07:56:52.9226214Z     logger.go:130: 2022-04-05T07:56:33.088Z	DEBUG	done processing headers	{"headerIndex": 37, "blockHeight": 36, "took": "533.306µs"}
2022-04-05T07:56:52.9227073Z     logger.go:130: 2022-04-05T07:56:33.088Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9228009Z     logger.go:130: 2022-04-05T07:56:33.088Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9228867Z     logger.go:130: 2022-04-05T07:56:33.089Z	DEBUG	done processing headers	{"headerIndex": 38, "blockHeight": 37, "took": "541.305µs"}
2022-04-05T07:56:52.9229723Z     logger.go:130: 2022-04-05T07:56:33.089Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9230583Z     logger.go:130: 2022-04-05T07:56:33.089Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9232632Z     logger.go:130: 2022-04-05T07:56:33.089Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9234477Z     logger.go:130: 2022-04-05T07:56:33.089Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9236311Z     logger.go:130: 2022-04-05T07:56:33.090Z	DEBUG	done processing headers	{"headerIndex": 39, "blockHeight": 38, "took": "490.805µs"}
2022-04-05T07:56:52.9240033Z     logger.go:130: 2022-04-05T07:56:33.091Z	DEBUG	done processing headers	{"headerIndex": 40, "blockHeight": 39, "took": "485.205µs"}
2022-04-05T07:56:52.9241841Z     logger.go:130: 2022-04-05T07:56:33.091Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9243645Z     logger.go:130: 2022-04-05T07:56:33.091Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9263962Z     logger.go:130: 2022-04-05T07:56:33.092Z	DEBUG	done processing headers	{"headerIndex": 41, "blockHeight": 40, "took": "505.306µs"}
2022-04-05T07:56:52.9264838Z     logger.go:130: 2022-04-05T07:56:33.092Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9265520Z     logger.go:130: 2022-04-05T07:56:33.092Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9266460Z     logger.go:130: 2022-04-05T07:56:33.092Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9267120Z     logger.go:130: 2022-04-05T07:56:33.092Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9272674Z     logger.go:130: 2022-04-05T07:56:33.093Z	DEBUG	done processing headers	{"headerIndex": 42, "blockHeight": 41, "took": "503.505µs"}
2022-04-05T07:56:52.9273430Z     logger.go:130: 2022-04-05T07:56:33.094Z	DEBUG	done processing headers	{"headerIndex": 43, "blockHeight": 42, "took": "544.606µs"}
2022-04-05T07:56:52.9274102Z     logger.go:130: 2022-04-05T07:56:33.094Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9275089Z     logger.go:130: 2022-04-05T07:56:33.094Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9275922Z     logger.go:130: 2022-04-05T07:56:33.095Z	DEBUG	done processing headers	{"headerIndex": 44, "blockHeight": 43, "took": "490.606µs"}
2022-04-05T07:56:52.9276583Z     logger.go:130: 2022-04-05T07:56:33.095Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9277242Z     logger.go:130: 2022-04-05T07:56:33.095Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9277902Z     logger.go:130: 2022-04-05T07:56:33.095Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9278563Z     logger.go:130: 2022-04-05T07:56:33.095Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9279182Z     logger.go:130: 2022-04-05T07:56:33.096Z	DEBUG	done processing headers	{"headerIndex": 45, "blockHeight": 44, "took": "525.206µs"}
2022-04-05T07:56:52.9279846Z     logger.go:130: 2022-04-05T07:56:33.097Z	DEBUG	done processing headers	{"headerIndex": 46, "blockHeight": 45, "took": "489.805µs"}
2022-04-05T07:56:52.9280498Z     logger.go:130: 2022-04-05T07:56:33.097Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9281148Z     logger.go:130: 2022-04-05T07:56:33.097Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9282022Z     logger.go:130: 2022-04-05T07:56:33.098Z	DEBUG	done processing headers	{"headerIndex": 47, "blockHeight": 46, "took": "487.405µs"}
2022-04-05T07:56:52.9282722Z     logger.go:130: 2022-04-05T07:56:33.098Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9283374Z     logger.go:130: 2022-04-05T07:56:33.098Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9284031Z     logger.go:130: 2022-04-05T07:56:33.098Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9284681Z     logger.go:130: 2022-04-05T07:56:33.098Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9285299Z     logger.go:130: 2022-04-05T07:56:33.099Z	DEBUG	done processing headers	{"headerIndex": 48, "blockHeight": 47, "took": "490.206µs"}
2022-04-05T07:56:52.9285916Z     logger.go:130: 2022-04-05T07:56:33.100Z	DEBUG	done processing headers	{"headerIndex": 49, "blockHeight": 48, "took": "493.505µs"}
2022-04-05T07:56:52.9286571Z     logger.go:130: 2022-04-05T07:56:33.100Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9287202Z     logger.go:130: 2022-04-05T07:56:33.100Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9288295Z     logger.go:130: 2022-04-05T07:56:33.100Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9289023Z     logger.go:130: 2022-04-05T07:56:33.100Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9289746Z     logger.go:130: 2022-04-05T07:56:33.100Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9290403Z     logger.go:130: 2022-04-05T07:56:33.100Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9291051Z     logger.go:130: 2022-04-05T07:56:33.100Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9291697Z     logger.go:130: 2022-04-05T07:56:33.100Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9292350Z     logger.go:130: 2022-04-05T07:56:33.100Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9292993Z     logger.go:130: 2022-04-05T07:56:33.100Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9293625Z     logger.go:130: 2022-04-05T07:56:33.101Z	DEBUG	done processing headers	{"headerIndex": 50, "blockHeight": 49, "took": "550.006µs"}
2022-04-05T07:56:52.9294240Z     logger.go:130: 2022-04-05T07:56:33.357Z	DEBUG	done processing headers	{"headerIndex": 51, "blockHeight": 50, "took": "510.405µs"}
2022-04-05T07:56:52.9294890Z     logger.go:130: 2022-04-05T07:56:33.357Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9295877Z     logger.go:130: 2022-04-05T07:56:33.358Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9296541Z     logger.go:130: 2022-04-05T07:56:33.358Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9297165Z     logger.go:130: 2022-04-05T07:56:33.359Z	DEBUG	done processing headers	{"headerIndex": 52, "blockHeight": 51, "took": "571.506µs"}
2022-04-05T07:56:52.9297811Z     logger.go:130: 2022-04-05T07:56:33.359Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9298601Z     logger.go:130: 2022-04-05T07:56:33.359Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9299255Z     logger.go:130: 2022-04-05T07:56:33.360Z	DEBUG	done processing headers	{"headerIndex": 53, "blockHeight": 52, "took": "563.906µs"}
2022-04-05T07:56:52.9299907Z     logger.go:130: 2022-04-05T07:56:33.360Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9300560Z     logger.go:130: 2022-04-05T07:56:33.360Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9301178Z     logger.go:130: 2022-04-05T07:56:33.361Z	DEBUG	done processing headers	{"headerIndex": 54, "blockHeight": 53, "took": "568.206µs"}
2022-04-05T07:56:52.9301824Z     logger.go:130: 2022-04-05T07:56:33.362Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9302469Z     logger.go:130: 2022-04-05T07:56:33.362Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9303091Z     logger.go:130: 2022-04-05T07:56:33.362Z	DEBUG	done processing headers	{"headerIndex": 55, "blockHeight": 54, "took": "558.506µs"}
2022-04-05T07:56:52.9303739Z     logger.go:130: 2022-04-05T07:56:33.363Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9304532Z     logger.go:130: 2022-04-05T07:56:33.363Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9305155Z     logger.go:130: 2022-04-05T07:56:33.364Z	DEBUG	done processing headers	{"headerIndex": 56, "blockHeight": 55, "took": "558.106µs"}
2022-04-05T07:56:52.9305804Z     logger.go:130: 2022-04-05T07:56:33.364Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9306447Z     logger.go:130: 2022-04-05T07:56:33.364Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9307071Z     logger.go:130: 2022-04-05T07:56:33.365Z	DEBUG	done processing headers	{"headerIndex": 57, "blockHeight": 56, "took": "561.006µs"}
2022-04-05T07:56:52.9307719Z     logger.go:130: 2022-04-05T07:56:33.365Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9308374Z     logger.go:130: 2022-04-05T07:56:33.365Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9308990Z     logger.go:130: 2022-04-05T07:56:33.366Z	DEBUG	done processing headers	{"headerIndex": 58, "blockHeight": 57, "took": "556.706µs"}
2022-04-05T07:56:52.9309635Z     logger.go:130: 2022-04-05T07:56:33.366Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9310280Z     logger.go:130: 2022-04-05T07:56:33.367Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9310899Z     logger.go:130: 2022-04-05T07:56:33.367Z	DEBUG	done processing headers	{"headerIndex": 59, "blockHeight": 58, "took": "558.806µs"}
2022-04-05T07:56:52.9311548Z     logger.go:130: 2022-04-05T07:56:33.368Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9312203Z     logger.go:130: 2022-04-05T07:56:33.368Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9312798Z     logger.go:130: 2022-04-05T07:56:33.369Z	DEBUG	done processing headers	{"headerIndex": 60, "blockHeight": 59, "took": "556.006µs"}
2022-04-05T07:56:52.9313445Z     logger.go:130: 2022-04-05T07:56:33.369Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9314088Z     logger.go:130: 2022-04-05T07:56:33.369Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9319674Z     logger.go:130: 2022-04-05T07:56:33.370Z	DEBUG	done processing headers	{"headerIndex": 61, "blockHeight": 60, "took": "559.206µs"}
2022-04-05T07:56:52.9320465Z     logger.go:130: 2022-04-05T07:56:33.370Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9321140Z     logger.go:130: 2022-04-05T07:56:33.370Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9321768Z     logger.go:130: 2022-04-05T07:56:33.371Z	DEBUG	done processing headers	{"headerIndex": 62, "blockHeight": 61, "took": "561.706µs"}
2022-04-05T07:56:52.9322427Z     logger.go:130: 2022-04-05T07:56:33.372Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9323083Z     logger.go:130: 2022-04-05T07:56:33.372Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9323713Z     logger.go:130: 2022-04-05T07:56:33.372Z	DEBUG	done processing headers	{"headerIndex": 63, "blockHeight": 62, "took": "559.706µs"}
2022-04-05T07:56:52.9324365Z     logger.go:130: 2022-04-05T07:56:33.373Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9325154Z     logger.go:130: 2022-04-05T07:56:33.373Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9325776Z     logger.go:130: 2022-04-05T07:56:33.374Z	DEBUG	done processing headers	{"headerIndex": 64, "blockHeight": 63, "took": "559.606µs"}
2022-04-05T07:56:52.9326426Z     logger.go:130: 2022-04-05T07:56:33.374Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9327079Z     logger.go:130: 2022-04-05T07:56:33.374Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9327703Z     logger.go:130: 2022-04-05T07:56:33.375Z	DEBUG	done processing headers	{"headerIndex": 65, "blockHeight": 64, "took": "559.006µs"}
2022-04-05T07:56:52.9328357Z     logger.go:130: 2022-04-05T07:56:33.375Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9329013Z     logger.go:130: 2022-04-05T07:56:33.375Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9329626Z     logger.go:130: 2022-04-05T07:56:33.376Z	DEBUG	done processing headers	{"headerIndex": 66, "blockHeight": 65, "took": "555.006µs"}
2022-04-05T07:56:52.9330277Z     logger.go:130: 2022-04-05T07:56:33.376Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9330930Z     logger.go:130: 2022-04-05T07:56:33.377Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9331629Z     logger.go:130: 2022-04-05T07:56:33.377Z	DEBUG	done processing headers	{"headerIndex": 67, "blockHeight": 66, "took": "566.006µs"}
2022-04-05T07:56:52.9332288Z     logger.go:130: 2022-04-05T07:56:33.378Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9332943Z     logger.go:130: 2022-04-05T07:56:33.378Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9333561Z     logger.go:130: 2022-04-05T07:56:33.379Z	DEBUG	done processing headers	{"headerIndex": 68, "blockHeight": 67, "took": "553.606µs"}
2022-04-05T07:56:52.9334209Z     logger.go:130: 2022-04-05T07:56:33.379Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9334860Z     logger.go:130: 2022-04-05T07:56:33.379Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9336094Z     logger.go:130: 2022-04-05T07:56:33.380Z	DEBUG	done processing headers	{"headerIndex": 69, "blockHeight": 68, "took": "562.106µs"}
2022-04-05T07:56:52.9336781Z     logger.go:130: 2022-04-05T07:56:33.380Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9337440Z     logger.go:130: 2022-04-05T07:56:33.380Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9338060Z     logger.go:130: 2022-04-05T07:56:33.381Z	DEBUG	done processing headers	{"headerIndex": 70, "blockHeight": 69, "took": "559.206µs"}
2022-04-05T07:56:52.9338709Z     logger.go:130: 2022-04-05T07:56:33.382Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9339354Z     logger.go:130: 2022-04-05T07:56:33.382Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9340007Z     logger.go:130: 2022-04-05T07:56:33.382Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9340654Z     logger.go:130: 2022-04-05T07:56:33.382Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9341399Z     logger.go:130: 2022-04-05T07:56:33.383Z	DEBUG	done processing headers	{"headerIndex": 71, "blockHeight": 70, "took": "840.71µs"}
2022-04-05T07:56:52.9342055Z     logger.go:130: 2022-04-05T07:56:33.383Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9342685Z     logger.go:130: 2022-04-05T07:56:33.384Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9343334Z     logger.go:130: 2022-04-05T07:56:33.384Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9343986Z     logger.go:130: 2022-04-05T07:56:33.384Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9344606Z     logger.go:130: 2022-04-05T07:56:33.536Z	DEBUG	done processing headers	{"headerIndex": 72, "blockHeight": 71, "took": "539.006µs"}
2022-04-05T07:56:52.9345291Z     logger.go:130: 2022-04-05T07:56:33.557Z	INFO	persisted to disk	{"blocks": 71, "keys": 828, "headerHeight": 72, "blockHeight": 72, "took": "452.405µs"}
2022-04-05T07:56:52.9345913Z     logger.go:130: 2022-04-05T07:56:33.792Z	DEBUG	done processing headers	{"headerIndex": 73, "blockHeight": 72, "took": "515.905µs"}
2022-04-05T07:56:52.9346564Z     logger.go:130: 2022-04-05T07:56:33.792Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9347209Z     logger.go:130: 2022-04-05T07:56:33.793Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9347862Z     logger.go:130: 2022-04-05T07:56:33.793Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9348481Z     logger.go:130: 2022-04-05T07:56:33.793Z	DEBUG	done processing headers	{"headerIndex": 74, "blockHeight": 73, "took": "396.404µs"}
2022-04-05T07:56:52.9349129Z     logger.go:130: 2022-04-05T07:56:33.794Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9349784Z     logger.go:130: 2022-04-05T07:56:33.794Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9350406Z     logger.go:130: 2022-04-05T07:56:33.794Z	DEBUG	done processing headers	{"headerIndex": 75, "blockHeight": 74, "took": "387.704µs"}
2022-04-05T07:56:52.9351048Z     logger.go:130: 2022-04-05T07:56:33.794Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9352519Z     logger.go:130: 2022-04-05T07:56:33.795Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9353252Z     logger.go:130: 2022-04-05T07:56:33.795Z	DEBUG	done processing headers	{"headerIndex": 76, "blockHeight": 75, "took": "357.004µs"}
2022-04-05T07:56:52.9353904Z     logger.go:130: 2022-04-05T07:56:33.795Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9354564Z     logger.go:130: 2022-04-05T07:56:33.796Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9355272Z     logger.go:130: 2022-04-05T07:56:33.796Z	DEBUG	done processing headers	{"headerIndex": 77, "blockHeight": 76, "took": "395.505µs"}
2022-04-05T07:56:52.9355920Z     logger.go:130: 2022-04-05T07:56:33.796Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9356574Z     logger.go:130: 2022-04-05T07:56:33.796Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9357191Z     logger.go:130: 2022-04-05T07:56:33.797Z	DEBUG	done processing headers	{"headerIndex": 78, "blockHeight": 77, "took": "361.604µs"}
2022-04-05T07:56:52.9357841Z     logger.go:130: 2022-04-05T07:56:33.797Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9358610Z     logger.go:130: 2022-04-05T07:56:33.797Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9359232Z     logger.go:130: 2022-04-05T07:56:33.798Z	DEBUG	done processing headers	{"headerIndex": 79, "blockHeight": 78, "took": "318.403µs"}
2022-04-05T07:56:52.9359880Z     logger.go:130: 2022-04-05T07:56:33.798Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9360536Z     logger.go:130: 2022-04-05T07:56:33.798Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9361154Z     logger.go:130: 2022-04-05T07:56:33.798Z	DEBUG	done processing headers	{"headerIndex": 80, "blockHeight": 79, "took": "331.704µs"}
2022-04-05T07:56:52.9361803Z     logger.go:130: 2022-04-05T07:56:33.798Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9362457Z     logger.go:130: 2022-04-05T07:56:33.799Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9363075Z     logger.go:130: 2022-04-05T07:56:33.799Z	DEBUG	done processing headers	{"headerIndex": 81, "blockHeight": 80, "took": "438.305µs"}
2022-04-05T07:56:52.9363730Z     logger.go:130: 2022-04-05T07:56:33.799Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9364382Z     logger.go:130: 2022-04-05T07:56:33.800Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9364995Z     logger.go:130: 2022-04-05T07:56:33.800Z	DEBUG	done processing headers	{"headerIndex": 82, "blockHeight": 81, "took": "450.105µs"}
2022-04-05T07:56:52.9365642Z     logger.go:130: 2022-04-05T07:56:33.800Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9402362Z     logger.go:130: 2022-04-05T07:56:33.800Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9403609Z     logger.go:130: 2022-04-05T07:56:33.801Z	DEBUG	done processing headers	{"headerIndex": 83, "blockHeight": 82, "took": "396.604µs"}
2022-04-05T07:56:52.9404525Z     logger.go:130: 2022-04-05T07:56:33.801Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9405593Z     logger.go:130: 2022-04-05T07:56:33.801Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9406573Z     logger.go:130: 2022-04-05T07:56:33.802Z	DEBUG	done processing headers	{"headerIndex": 84, "blockHeight": 83, "took": "324.103µs"}
2022-04-05T07:56:52.9407481Z     logger.go:130: 2022-04-05T07:56:33.802Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9408478Z     logger.go:130: 2022-04-05T07:56:33.802Z	DEBUG	done processing headers	{"headerIndex": 85, "blockHeight": 84, "took": "289.003µs"}
2022-04-05T07:56:52.9409357Z     logger.go:130: 2022-04-05T07:56:33.802Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9410231Z     logger.go:130: 2022-04-05T07:56:33.803Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9411108Z     logger.go:130: 2022-04-05T07:56:33.803Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9411949Z     logger.go:130: 2022-04-05T07:56:33.803Z	DEBUG	done processing headers	{"headerIndex": 86, "blockHeight": 85, "took": "444.705µs"}
2022-04-05T07:56:52.9412774Z     logger.go:130: 2022-04-05T07:56:33.804Z	DEBUG	done processing headers	{"headerIndex": 87, "blockHeight": 86, "took": "770.408µs"}
2022-04-05T07:56:52.9413857Z     logger.go:130: 2022-04-05T07:56:33.805Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9414861Z     logger.go:130: 2022-04-05T07:56:33.805Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9416015Z     logger.go:130: 2022-04-05T07:56:33.805Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9416899Z     logger.go:130: 2022-04-05T07:56:33.805Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9417748Z     logger.go:130: 2022-04-05T07:56:33.805Z	DEBUG	done processing headers	{"headerIndex": 88, "blockHeight": 87, "took": "362.504µs"}
2022-04-05T07:56:52.9418609Z     logger.go:130: 2022-04-05T07:56:33.806Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9419483Z     logger.go:130: 2022-04-05T07:56:33.806Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9420320Z     logger.go:130: 2022-04-05T07:56:33.806Z	DEBUG	done processing headers	{"headerIndex": 89, "blockHeight": 88, "took": "385.904µs"}
2022-04-05T07:56:52.9421176Z     logger.go:130: 2022-04-05T07:56:33.806Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9422038Z     logger.go:130: 2022-04-05T07:56:33.806Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9422882Z     logger.go:130: 2022-04-05T07:56:33.807Z	DEBUG	done processing headers	{"headerIndex": 90, "blockHeight": 89, "took": "381.904µs"}
2022-04-05T07:56:52.9423754Z     logger.go:130: 2022-04-05T07:56:33.807Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9424733Z     logger.go:130: 2022-04-05T07:56:33.807Z	DEBUG	done processing headers	{"headerIndex": 91, "blockHeight": 90, "took": "299.703µs"}
2022-04-05T07:56:52.9425557Z     logger.go:130: 2022-04-05T07:56:33.808Z	DEBUG	done processing headers	{"headerIndex": 92, "blockHeight": 91, "took": "526.406µs"}
2022-04-05T07:56:52.9426440Z     logger.go:130: 2022-04-05T07:56:33.807Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9427301Z     logger.go:130: 2022-04-05T07:56:33.809Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9428319Z     logger.go:130: 2022-04-05T07:56:33.809Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9430059Z     logger.go:130: 2022-04-05T07:56:33.809Z	DEBUG	done processing headers	{"headerIndex": 93, "blockHeight": 92, "took": "525.506µs"}
2022-04-05T07:56:52.9433153Z     logger.go:130: 2022-04-05T07:56:33.810Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9435009Z     logger.go:130: 2022-04-05T07:56:33.810Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9437864Z     logger.go:130: 2022-04-05T07:56:33.810Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9439667Z     logger.go:130: 2022-04-05T07:56:33.810Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2022-04-05T07:56:52.9445969Z     logger.go:130: 2022-04-05T07:56:34.557Z	INFO	persisted to disk	{"blocks": 21, "keys": 262, "headerHeight": 93, "blockHeight": 93, "took": "116.101µs"}
2022-04-05T07:56:52.9446168Z     notary_test.go:613: 
2022-04-05T07:56:52.9448056Z         	Error Trace:	notary_test.go:613
2022-04-05T07:56:52.9448544Z         	Error:      	Condition never satisfied
2022-04-05T07:56:52.9448755Z         	Test:       	TestNotary
2022-04-05T07:56:52.9448998Z --- FAIL: TestNotary (6.26s)
```

--- a/pkg/services/notary/core_test.go
+++ b/pkg/services/notary/core_test.go
@@ -133,7 +133,7 @@ func TestNotary(t *testing.T) {
 			defer mtx.RUnlock()
 			completedTx = completedTxes[h]
 			return completedTx != nil
-		}, time.Second*3, time.Millisecond*50, errors.New("main transaction expected to be completed"))
+		}, time.Second*3, time.Millisecond*50, errors.New("transaction expected to be completed"))
 		return completedTx
 	}
 

--- a/pkg/services/notary/core_test.go
+++ b/pkg/services/notary/core_test.go
@@ -600,22 +600,14 @@ func TestNotary(t *testing.T) {
 	// Add block before allowing tx to finalize to exclude race condition when
 	// main transaction is finalized between `finalizeWithError` restore and adding new block.
 	e.AddNewBlock(t)
-	mtx.RLock()
-	start := len(completedTxes)
-	mtx.RUnlock()
 	setFinalizeWithError(false)
-	for i := range requests {
-		if i != 0 {
-			e.AddNewBlock(t)
-		}
-		require.Eventually(t, func() bool {
-			mtx.RLock()
-			defer mtx.RUnlock()
-			return len(completedTxes)-start >= i+1
-		}, time.Second*3, time.Millisecond)
+	for i := range requests[1:] {
+		e.AddNewBlock(t)
+
+		_ = getCompletedTx(t, true, requests[i+1].FallbackTransaction.Hash())
 		checkMainTx(t, requesters, requests, len(requests), false)
-		checkFallbackTxs(t, requests[:i+1], true)
-		checkFallbackTxs(t, requests[i+1:], false)
+		checkFallbackTxs(t, requests[:i+2], true)
+		checkFallbackTxs(t, requests[i+2:], false)
 	}
 
 	// OnRequestRemoval: missing account

